app/org/sagebionetworks/bridge/services/NotificationsService.java

```diff
@@ -106,6 +128,45 @@ public void deleteRegistration(String healthCode, String guid) {
         
         notificationRegistrationDao.deleteRegistration(healthCode, guid);
     }
+    
+    /**
+     * Send a push notification to an individual participant (assuming they have registered for push notification). 
+     * This mechanism is intended to message specific individuals, <i>and should not be used to send out notifications 
+     * to many accounts.</i> Create a topic, ask your users to subscribe to that topic in your application, and message 
+     * them via that topic.
+     */
+    public void sendNotification(StudyIdentifier studyId, String healthCode, NotificationMessage message) {
+        checkNotNull(studyId);
+        checkNotNull(healthCode);
+        checkNotNull(message);
+        
+        Validate.entityThrowingException(NotificationMessageValidator.INSTANCE, message);
+        
+        List<NotificationRegistration> registrations = notificationRegistrationDao.listRegistrations(healthCode);
+        if (registrations.isEmpty()) {
+            throw new BadRequestException("Participant has not registered to receive push notifications.");
```

Comment:
In the case where there are no registrations, we know for a fact it will fail, so right now the UI disables the button to send a notification to the user. I think we should probably return 400 here to tell the caller since we know it's pointless. In the case where it succeeds though, we should return 202 "Accepted" since actually, we don't know what the ultimate disposition of the message will be. Even without errors, it may fail to be delivered for any number of reasons. So I'll change that.